### PR TITLE
Libhugetlbfs

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/libhugetlbfs/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - test -f $PREFIX/lib/libhugetlbfs${SHLIB_EXT}
+    - hugectl --help
+    - hugeadm --help
+    - hugeedit --help
+
+about:
+  home: https://github.com/libhugetlbfs/libhugetlbfs
+  # Remember to specify the license variants for BSD, Apache, GPL, and LGPL.
+  # Use the SPDX identifier, e.g: GPL-2.0-only instead of GNU General Public License version 2.0
+  # See https://spdx.org/licenses/
```
Comment (jakirkham) [inline recipes/libhugetlbfs/meta.yaml:32]: This can be dropped

```suggestion
```

File: recipes/libhugetlbfs/build.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+if [[ ${ARCH} == "64" ]]; then
+    export ARCH=x86_64
+fi
+
+# Remove the static library from being installed
+sed -i 's/libhugetlbfs.a//g' Makefile
```
Comment (jakirkham) [inline recipes/libhugetlbfs/build.sh:6]: Should we also test to see this is removed?

File: recipes/libhugetlbfs/build.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+if [[ ${ARCH} == "64" ]]; then
+    export ARCH=x86_64
+fi
```
Comment (jakirkham) [inline recipes/libhugetlbfs/build.sh:3]: Why is this needed? Maybe worth including a comment if this is something the `Makefile` is using

File: recipes/libhugetlbfs/build.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+# Remove the static library from being installed
+sed -i 's/libhugetlbfs.a//g' Makefile
+
+# We only bulid native, so we want this to go to the correct location
+# The location is lib, nomatter what
+export LIB64=lib
+export LIB32=lib
```
Comment (jakirkham) [inline recipes/libhugetlbfs/build.sh:11]: Should these be in `PREFIX` or are they already prefaced with this?

```suggestion
export LIB64="${PREFIX}/lib"
export LIB32="${PREFIX}/lib"
```

File: recipes/libhugetlbfs/build.sh [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+export BUILDTYPE=NATIVEONLY
+# Make library and tools, but not tests
+make libs tools -j${CPU_COUNT}
+
+# Do not install man pages
+make install-libs install-bin
```
Comment (jakirkham) [inline recipes/libhugetlbfs/build.sh:18]: Do we need to specify `PREFIX` with either of these? Even if not, it might be good to include it (and `ARCH` depending on previous comment) for clarity so future readers know the `Makefile` is using these.

**Review by jakirkham — COMMENTED** [review]
Thanks Mark! 😄

Had a couple questions/suggestions below :)

File: recipes/libhugetlbfs/build.sh (hunk 3 above)
Comment (hmaarrfk) [reply recipes/libhugetlbfs/build.sh:3]: because the default makefile tries to check if ARCH is a known value and changes what it builds for it.

File: recipes/libhugetlbfs/build.sh (hunk 4 above)
Comment (hmaarrfk) [reply recipes/libhugetlbfs/build.sh:11]: From the makefile
```
REALLIB32 = $(realpath $(PREFIX)/$(LIB32))
REALLIB64 = $(realpath $(PREFIX)/$(LIB64))
```

```
LIBDIR32 = $(PREFIX)/$(LIB32)
LIBDIR64 = $(PREFIX)/$(LIB64)
```


```
obj32/install:
	@$(VECHO) INSTALL-LIB32 $(LIBDIR32)
	$(INSTALL) -d $(DESTDIR)$(LIBDIR32)
	$(INSTALL) $(INSTALL_OBJ_LIBS:%=obj32/%) $(DESTDIR)$(LIBDIR32)

obj64/install:
	@$(VECHO) INSTALL-LIB64 $(LIBDIR64)
	$(INSTALL) -d $(DESTDIR)$(LIBDIR64)
	$(INSTALL) $(INSTALL_OBJ_LIBS:%=obj64/%) $(DESTDIR)$(LIBDIR64)

```

File: recipes/libhugetlbfs/build.sh (hunk 2 above)
Comment (hmaarrfk) [reply recipes/libhugetlbfs/build.sh:9]: done.

File: recipes/libhugetlbfs/build.sh (hunk 5 above)
Comment (hmaarrfk) [reply recipes/libhugetlbfs/build.sh:22]: done.

**Comment by hmaarrfk** [comment]
Thanks for the review.

Comments greatly expanded.

**Comment by hmaarrfk** [comment]
@jakirkham are you able to merge and approve this?

**Comment by hmaarrfk** [comment]
@jakirkham kind ping here!

**Comment by hmaarrfk** [comment]
@jakirkham kind ping here!

@conda-forge/staged-recipes this is ready for review.

File: recipes/libhugetlbfs/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/libhugetlbfs/libhugetlbfs/releases/download/{{ version }}/libhugetlbfs-{{ version }}.tar.gz
+  sha256: b70672f1e807c61b7eb3adf41c1903b42917951f2e7f8aef6821841700c04479
+
+build:
```
Comment (xhochy) [inline recipes/libhugetlbfs/meta.yaml:12]: Should there be a run_exports here?

Comment (hmaarrfk) [reply recipes/libhugetlbfs/meta.yaml:11]: I added one thanks!

Comment (hmaarrfk) [reply recipes/libhugetlbfs/meta.yaml:11]: Is there a way to automatically launch migrations?

Comment (xhochy) [reply recipes/libhugetlbfs/meta.yaml:11]: The bot will make automatic PRs on a new version to conda-forge-pinning if it is listed in the global pinning.
